distrho/src/DistrhoPluginAU.cpp

```diff
@@ -182,43 +180,33 @@ class PluginAU : public AUEffectBase
         return fPlugin.getVersion();
     }
 
-    // -------------------------------------------------------------------
-
-private:
-    LastValuesInit fLastValuesInit;
-    PluginExporter fPlugin;
-
-    // most of the real work happens here
-    class PluginKernel : public AUKernelBase
+    OSStatus ProcessBufferLists(AudioUnitRenderActionFlags &ioActionFlags,
+                                const AudioBufferList &inBuffer,
+                                AudioBufferList &outBuffer,
+                                UInt32 inFramesToProcess) override
     {
-    public:
-        PluginKernel(AUEffectBase* const inAudioUnit, PluginExporter& plugin)
-            : AUKernelBase(inAudioUnit),
-              fPlugin(plugin) {}
-
-        // *Required* overides for the process method for this effect
-        // processes one channel of interleaved samples
-        void Process(const Float32* const inSourceP,
-                     Float32* const inDestP,
-                     UInt32 inFramesToProcess,
-                     UInt32 inNumChannels,
-                     bool& ioSilence)
-        {
-            DISTRHO_SAFE_ASSERT_RETURN(inNumChannels == DISTRHO_PLUGIN_NUM_INPUTS,);
+        UInt32 i;
+        float *srcBuffer[maxch];
```

Comment:
srcBuffer should be a `const float` pointer, so you avoid doing type conversions during the run() function.
this dynamic channel looks like it will be a little pain to support...

Reply:
We still need to figure out how to create multiple instances of the plugin for multi-mono.